risc0/zkvm/sdk/cpp/host/c_api.cpp

```diff
@@ -77,9 +78,13 @@ void risc0_string_free(risc0_string* str) {
   ffi_wrap_void(&err, [&] { delete str; });
 }
 
-risc0_prover* risc0_prover_new(risc0_error* err, const char* elf_path, const char* method_id_path) {
+risc0_prover* risc0_prover_new(risc0_error* err,
+                               const char* elf_path,
+                               const uint8_t* method_id_buf,
+                               const size_t method_id_len) {
```

Comment:
I don't think we need the length here because it's a constant size.

Reply:
That's how I had it originally but felt a little uneasy about it. Happy to remove it though - you've definitely got more expertise here.